Broken doc-test

### Describe the change

## Issue

The doc test on line 48 of `crates/net/downloaders/src/headers/task.rs` seems broken.

## Steps to reproduce

1.  `cargo clean`
2. `cargo test --doc --all`

This will output:
```
---- src/headers/task.rs - headers::task::TaskDownloader::spawn (line 41) stdout ----
error[E0282]: type annotations needed
  --> src/headers/task.rs:48:21
   |
10 |    let downloader = LinearDownloader::builder().build(
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^ cannot infer type of the type parameter `H` declared on the struct `LinearDownloader`
   |
help: consider specifying the generic argument
   |
10 |    let downloader = LinearDownloader::<H>::builder().build(
   |                                     +++++

error: aborting due to previous error

```

### Additional context

_No response_

--- a/crates/net/downloaders/src/headers/task.rs
+++ b/crates/net/downloaders/src/headers/task.rs
@@ -45,7 +45,7 @@ impl TaskDownloader {
     /// # use reth_interfaces::consensus::Consensus;
     /// # use reth_interfaces::p2p::headers::client::HeadersClient;
     /// # fn t<H: HeadersClient + 'static>(consensus:Arc<dyn Consensus>, client: Arc<H>) {
-    ///    let downloader = LinearDownloader::builder().build(
+    ///    let downloader = LinearDownloader::<H>::builder().build(
     ///        consensus,
     ///        client,
     ///        Default::default(),

